src/main/java/org/opensearch/sdk/NettyTransport.java

```diff
@@ -0,0 +1,120 @@
+     * @param threadPool  A thread pool to use.
+     * @return The configured Netty4Transport object.
+     */
+    public Netty4Transport getNetty4Transport(Settings settings, ThreadPool threadPool) {
+        NetworkService networkService = new NetworkService(Collections.emptyList());
+        PageCacheRecycler pageCacheRecycler = new PageCacheRecycler(settings);
+        IndicesModule indicesModule = new IndicesModule(Collections.emptyList());
+        SearchModule searchModule = new SearchModule(settings, Collections.emptyList());
+
+        List<NamedWriteableRegistry.Entry> namedWriteables = Stream.of(
+            NetworkModule.getNamedWriteables().stream(),
+            indicesModule.getNamedWriteables().stream(),
+            searchModule.getNamedWriteables().stream(),
+            ClusterModule.getNamedWriteables().stream()
+        ).flatMap(Function.identity()).collect(Collectors.toList());
+
+        final NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry(namedWriteables);
+
+        final CircuitBreakerService circuitBreakerService = new NoneCircuitBreakerService();
+
+        Netty4Transport transport = new Netty4Transport(
+            settings,
+            Version.CURRENT,
+            threadPool,
+            networkService,
+            pageCacheRecycler,
+            namedWriteableRegistry,
+            circuitBreakerService,
+            new SharedGroupFactory(settings)
+        );
+
+        return transport;
+    }
+
+    /**
+     * Initializes the TransportService object for this extension. This object will control communication between the extension and OpenSearch.
+     *
+     * @param settings  The transport settings to configure.
+     * @param extensionsRunner method to call
+     * @return The initialized TransportService object.
+     */
+    public TransportService initializeExtensionTransportService(Settings settings, ExtensionsRunner extensionsRunner) {
+
+        ThreadPool threadPool = new ThreadPool(settings);
+
+        Netty4Transport transport = getNetty4Transport(settings, threadPool);
+
+        // Stop any existing transport service
+        if (extensionsRunner.extensionTransportService != null) {
+            extensionsRunner.extensionTransportService.stop();
+        }
+
+        // create transport service
+        extensionsRunner.extensionTransportService = new TransportService(
+            settings,
+            transport,
+            threadPool,
+            NOOP_TRANSPORT_INTERCEPTOR,
+            boundAddress -> DiscoveryNode.createLocal(
+                Settings.builder().put(NODE_NAME_SETTING, settings.get(NODE_NAME_SETTING)).build(),
+                boundAddress.publishAddress(),
+                randomBase64UUID()
+            ),
+            null,
+            emptySet()
+        );
+        extensionsRunner.startTransportService(extensionsRunner.extensionTransportService);
+        return extensionsRunner.extensionTransportService;
```

Comment:
(Suggestion) We're never using this return value, could just make this method return `void`.